src/app/(pages)/groups/components/table/options/index.tsx

```diff
@@ -1,18 +1,32 @@
 import Options from "@/components/options";
 import { Group } from "@/models/groups";
+import { ScimReference } from "@/models/scim";
+import RemoveMembership from "./remove-membership-button";
 import AddSubgroupButton from "./add-subgroup-button";
 import DeleteGroupButton from "./delete-group-button";
+import { auth } from "@/auth";
 
 export type GroupOptionsProps = {
   group: Group;
+  userRef?: ScimReference;
 };
 
-export default function GroupOptions(props: Readonly<GroupOptionsProps>) {
-  const { group } = props;
+export default async function GroupOptions(
+  props: Readonly<GroupOptionsProps>
+) {
+  const { group, userRef } = props;
+  const session = await auth();
+  const isAdmin = session?.is_admin ?? false
```

Comment:
I would not call `await auth()` here, plus the if clause. 
Even though the call should be cached, remember that you are calling `auth()` and the if clause for EACH row.
I would simply pass the `isAdmin?: boolean` prop in the `GroupOptionsProps` interface.
In this way we move the `auth()` call to the parent component, calling it once for all the rows